lib/ews/soap/exchange_web_service.rb

```diff
@@ -168,6 +170,12 @@ def get_user_availability(opts)
       do_soap_request(req, response_class: EwsSoapFreeBusyResponse)
     end
 
+    def get_user_settings(opts)
+      ews_builder = EwsBuilder.new
+      req = build_get_user_settings_request_doc(builder: ews_builder, users: opts[:users], requested_settings: opts[:requested_settings])
```

Comment:
`build_soap!` actually pass the block so we could do just like https://github.com/BookingBug/Viewpoint/pull/6/files#diff-2a6570e4959240206219dd27427f0ebaR111

```ruby
req = build_soap! do |type, builder|
  # building the requests here
end
```